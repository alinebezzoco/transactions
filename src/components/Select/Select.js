import React from "react";
import PropTypes from 'prop-types';
import "./Select.css";

const Select = props => {
  return (
    <div className="form-group">
      <label htmlFor={props.name}> {props.title} </label>
      <select
        id={props.name}
        name={props.name}
        value={props.value}
        onChange={props.handleChange}
        className="form-control"
        required={props.required}
      >
        <option value="" disabled>
          {props.placeholder}
        </option>
        {props.options.map(option => {
          return (
            <option key={option} value={option} label={option}>
              {option}
            </option>
          );
        })}
      </select>
    </div>
  );
};

Select.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string, 
  value: PropTypes.string, 
  handleChange: PropTypes.func, 
  placeholder: PropTypes.string, 
  required: PropTypes.bool,
  option: PropTypes.string
};


export default Select;
