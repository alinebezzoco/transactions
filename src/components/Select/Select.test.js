import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Select from "./Select";

describe("Select Component", () => {
  beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  it("renders correctly", () => {
    const transactionOptions = ["Débito", "Crédito"]
    const component = shallow(
      <Select
      title={"Tipo de transação"}
      name={"transaction"}
      options={transactionOptions}
      value={"Débito"}
      placeholder={"Selecionar"}
      required={true}
    />
    );
    expect(component).toMatchSnapshot();
  });
});
