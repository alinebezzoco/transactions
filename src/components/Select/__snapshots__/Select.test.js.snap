// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component renders correctly 1`] = `ShallowWrapper {}`;
